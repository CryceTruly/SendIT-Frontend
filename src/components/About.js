import React, {
  Component
} from 'react'

export default class About extends Component {
  render() {
    return (
      <div>
      <h1> This is about </h1> </div>
    )
  }
}