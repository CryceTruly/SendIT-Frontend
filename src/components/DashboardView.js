import React from 'react'

function DashboardView() {
  return (
    <div>
        <h3>Stats</h3>

    </div>
  )
}

export default DashboardView
