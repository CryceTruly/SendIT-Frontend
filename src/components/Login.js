import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import LoginView from "../views/Login";
import { clearErrors } from "../actions/errorActions";
import { login } from "../actions/authActions";

export class Login extends Component {
  state = {
    email: "",
    password: "",
    msg: null,
  };

  componentDidUpdate(prevProps) {
    const { error } = this.props;
    if (error !== prevProps.error) {
      // Check for auth error
      /* istanbul ignore next */
      if (error.id === "LOGIN_FAIL") {
        this.setState({ msg: error.msg });
      } else {
        this.setState({ msg: null });
      }
    }
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onSubmit = (e) => {
    e.preventDefault();

    const { email, password } = this.state;

    // Create user data
    const newUser = {
      email,
      password,
    };

    // Attempt to login
    this.props.login(newUser, this.props);
  };

  render() {
    return (
      <LoginView
        msg={this.state.msg}
        onChange={this.onChange}
        onSubmit={this.onSubmit}
        errorMessage={this.props.error.msg}
        errorId={this.props.error.id}
        processing={this.props.error.processing}
      />
    );
  }
}
export const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
  error: state.errors,
  auth: state.auth,
});

export default connect(
  mapStateToProps,
  { login, clearErrors },
)(Login);
