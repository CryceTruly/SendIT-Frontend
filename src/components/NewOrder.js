import React, { Component } from 'react'
import {
    Button,
    Card,
    CardBody,
    CardTitle,
    Form,
    FormGroup,
    Label,
    Input,
    Alert,
    Row,
    Col
  } from 'reactstrap';
export default class NewOrder extends Component {
    state = {
        fullname: '',
        email: '',
        password: '',
        phone_number:'',
        username:'',
        msg: null
      };

  render() {
    return (
      <div>

         <Card>
        <CardBody>
          <CardTitle></CardTitle>

          <CardTitle><strong>Add a new Parcel Request.</strong></CardTitle>


            <Form onSubmit={this.onSubmit}>


{this.state.msg?<Alert color="danger">{this.state.msg}</Alert>:null}

            <Row>
              <div className="col-md-6">

              <FormGroup>
                <Label for='fullname'>Pickup Location</Label>
                <Input
                  type='text'
                  name='fullname'
                  id='fullname'
                  placeholder='Full Name' required
                  className='mb-3'
                  onChange={this.onChange}
                />
                </FormGroup>


<FormGroup>
                <Label for='email'>Destination address</Label>
                <Input
                  type='email'
                  name='email'
                  id='email' required
                  placeholder='Email'
                  className='mb-3'
                  onChange={this.onChange}/>
                 </FormGroup>
                <FormGroup>
<Label for='phone_number'>Recipient Phone Number</Label>
                <Input
                  type='text'
                  name='phone_number'
                  id='phone_number' required
                  placeholder='phone_number'
                  className='mb-3'
                  onChange={this.onChange}
                />
                </FormGroup>

              </div>
               <div className="col-md-6">
<FormGroup>
  <Label for='username'>Recipient Fullname</Label>
                <Input
                  type='text'
                  name='username'
                  id='username' required
                  placeholder=''
                  className='mb-3'
                  onChange={this.onChange}
                />
</FormGroup>
<FormGroup>
   <Label for='password'>Parcel Description</Label>
                <Input
                  type='password'
                  name='password' required
                  id='password'
                  placeholder=''
                  className='mb-3'
                  onChange={this.onChange}
                />



</FormGroup>

<FormGroup>

                  <Label for='password_comfirm'>Weight </Label>
                <Input
                  type='password_comfirm'
                  name='password'
                  id='password_comfirm' required
                  placeholder=''
                  className='mb-3'
                  onChange={this.onChange}
                />
</FormGroup>
 <Button color='dark' style={{ marginTop: '2rem' }} block>
                  Submit
                </Button>


               </div>


            </Row>




            </Form>
            </CardBody>
            </Card>



      </div>
    )
  }
}
