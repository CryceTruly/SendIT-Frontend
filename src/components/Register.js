import { connect } from "react-redux";
import React, { Component } from "react";
import { clearErrors } from "../actions/errorActions";
import { register } from "../actions/authActions";
import Register from "../views/Register";

export class RegisterComponent extends Component {
  state = {
    fullname: "",
    email: "",
    password: "",
    phone_number: "",
    username: "",
    msg: null,
  };

  componentDidUpdate(prevProps) {
    const { error } = this.props;
    if (error !== prevProps.error) {
      // Check for register error
      /* istanbul ignore next */
      if (error.id === "REGISTER_FAIL") {
        this.setState({ msg: error.msg });
      } else {
        this.setState({ msg: null });
      }
    }
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onSubmit = (e) => {
    e.preventDefault();
    const {
      username, fullname, phone_number, email, password, password_comfirm,
    } = this.state;
    // Create user object
    const newUser = {
      username,
      email,
      password,
      phone_number,
      fullname,

    };

    if (password !== password_comfirm) {
      this.setState({ msg: "Passwords donot match" });
    } else
    /* istanbul ignore next */{
      this.setState({ msg: null });
      this.props.register(newUser, this.props);
    }
  };

  render() {
    return (
      <Register
        processing={
          this.props.error.processing}
        msg={this.state.msg}
        onChange={this.onChange}
        onSubmit={this.onSubmit}
      />
    );
  }
}
export const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
  error: state.errors,
});

export default connect(mapStateToProps, { register, clearErrors })(RegisterComponent);
