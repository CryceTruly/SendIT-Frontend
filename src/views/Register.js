import React from "react";
import {
  Button,
  Card,
  CardBody,
  CardTitle,
  Form,
  FormGroup,
  Label,
  Input,
  Alert,
  Row,
} from "reactstrap";
import Spinner from "./Spinner";

const Register = (props) => {
  const {
    onSubmit, onChange, processing, msg,
  } = props;
  return (
    <div>
      <Card>
        <CardBody>
          <CardTitle />
          <CardTitle><strong>Register an Account.</strong></CardTitle>
          <Form onSubmit={onSubmit}>
            {msg ? <Alert color="danger">{msg}</Alert> : null}
            <Row>
              <div className="col-md-6">
                <FormGroup>
                  <Label for="fullname">Fullname</Label>
                  <Input type="text" name="fullname" id="fullname" placeholder="Full Name" required className="mb-3" onChange={onChange} />
                </FormGroup>
                <FormGroup>
                  <Label for="email">Email</Label>
                  <Input type="email" name="email" id="email" required placeholder="Email" className="mb-3" onChange={onChange} />
                </FormGroup>
                <FormGroup>
                  <Label for="phone_number">Phone Number</Label>
                  <Input type="text" name="phone_number" id="phone_number" required placeholder="phone_number" className="mb-3" onChange={onChange} />
                </FormGroup>
              </div>
              <div className="col-md-6">
                <FormGroup>
                  <Label for="username">Username</Label>
                  <Input type="text" name="username" id="username" required className="mb-3" onChange={onChange} />
                </FormGroup>
                <FormGroup>
                  <Label for="password">Password</Label>
                  <Input type="password" name="password" required id="password" className="mb-3" onChange={onChange} />
                </FormGroup>
                <FormGroup>
                  <Label for="password_comfirm">Confirm Password </Label>
                  <Input type="password" name="password_comfirm" id="password_comfirm" required placeholder="" className="mb-3" onChange={onChange} />
                  {processing ? (
                    <Spinner />
                  ) : null
                  }
                </FormGroup>
                <Button disabled={processing} color="dark" style={{ marginTop: "2rem" }} block>
                  {!processing ? "Register" : "Please wait..."}
                </Button>
              </div>
            </Row>
          </Form>
        </CardBody>
      </Card>
    </div>
  );
};

export default Register;
