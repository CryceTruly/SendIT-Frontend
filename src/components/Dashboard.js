import React, { Component } from "react";
import DashboardView from "./DashboardView";

export default class Dashboard extends Component {
  render() {
    return (

      <h3>Welcome to SendIT Adminsration</h3>

    );
  }
}
