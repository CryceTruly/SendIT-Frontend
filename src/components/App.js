import React, {
    Component
} from "react";
import {
    Provider
} from "react-redux";
import {
    Container
} from "reactstrap";
import {
    BrowserRouter as Router, Switch, Link, Route
} from "react-router-dom";
import store from "../Store";
import Parcels from "./Parcels";
import Navbar from "./shared/Navbar";
import {
    loadUser
} from "../actions/authActions";
import Login from "./Login";
import RegisterComponent from "./Register";
import PrivateRoute from "./common/PrivateRoute";

export class App extends Component {

    componentDidMount(){
        store.dispatch(loadUser());
    }

    render() {
        return (
            <Router>
                <Provider store={store}>

                    <Navbar />
                    <Container>
                        <p />

                        <PrivateRoute exact path="/" component={Parcels} />
                        <Route exact path="/login" component={Login} />
                        <Route exact path="/parcels" component={Parcels} />
                        <Route exact path="/register" component={RegisterComponent} />
                    </Container>
                </Provider>
            </Router>





        );
    }
}
export default App;
