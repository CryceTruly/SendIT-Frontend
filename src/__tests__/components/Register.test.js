
import React from "react";
import { shallow } from "enzyme";
import { RegisterComponent, mapStateToProps } from "../../components/Register";


describe("registration tests", () => {
  const props = {
    error: {
      errorId: 1, errorMessage: "hi", processing: true, msg: "",
    },
    onChange: jest.fn(),
    onSubmit: jest.fn(),
    login: jest.fn(),
  };

  const wrapper = shallow(<RegisterComponent {...props} />);

  it("renders without crashing", () => {
    const e = {
      target: {
        name: "hey", value: "How are you",
      },
    };
    wrapper.instance().onChange(e);
    expect(wrapper.instance().state.hey).toEqual("How are you");
    expect(wrapper).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const wrapper = shallow(<RegisterComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });


  it("should submit a form", () => {
    const e = {
      preventDefault: () => {
      },
      target: {
        password: "hey",
        value: "password",
        password_comfirm: "hey2",
        value: "password",
      },
    };
    wrapper.instance().onSubmit(e);
  });

  it("should submit a form  on match", () => {
    const e = {
      preventDefault: () => {
      },
      target: {
        password: "hey",
        value: "password",
        password_comfirm: "hey",
        value: "password",
      },
    };
    wrapper.instance().onSubmit(e);
  });

  it("should map state to props", () => {
    const auth = {
      auth: {
        isAuthenticated: false,
        user: [],
        user_id: localStorage.getItem("user_id"),
        isLoading: false,
      },
    };
    const appState = auth;
    expect(mapStateToProps(appState)).toBeDefined();
  });
});
