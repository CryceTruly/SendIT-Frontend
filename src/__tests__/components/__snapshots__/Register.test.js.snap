// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registration tests renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegisterComponent
    error={
      Object {
        "errorId": 1,
        "errorMessage": "hi",
        "msg": "",
        "processing": true,
      }
    }
    login={[MockFunction]}
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "msg": null,
      "onChange": [Function],
      "onSubmit": [Function],
      "processing": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "msg": null,
        "onChange": [Function],
        "onSubmit": [Function],
        "processing": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`registration tests renders without crashing 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegisterComponent
    error={
      Object {
        "errorId": 1,
        "errorMessage": "hi",
        "msg": "",
        "processing": true,
      }
    }
    login={[MockFunction]}
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "msg": null,
      "onChange": [Function],
      "onSubmit": [Function],
      "processing": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "msg": null,
        "onChange": [Function],
        "onSubmit": [Function],
        "processing": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
